### PR TITLE
outdims: possible enhancements

### PR DESCRIPTION
**Pull request by hhaensel**
This PR is meant to discuss and develop further improvements to `outdims` as discussed in #1086 

I covered @HamletWantToCode 's proposal for a fallback (still without testing) and modified the dense version to keep all dimensions, as proposed by @darsnack .

Please feel free to also contribute commits.
Good night ;-)

### CODE REVIEW

**Comment by CarloLucibello** [comment]
please rebase on master, know that #1252 is merged

File: src/layers/basic.jl [hunk 1]
```diff
@@ -64,6 +64,10 @@ outdims(c::Chain, isize) = foldr(outdims, reverse(c.layers), init = isize)
 # see issue https://github.com/FluxML/Flux.jl/issues/702
 # Johnny Chen -- @johnnychen94
 # only slightly changed to better handle interaction with Zygote @dsweber2
+
+# fallback
+outdims(f::Function, isize::Tuple) = size(f(ones(Float32, isize..., 1)))[1:end-1]    # since we aren't care about batch dimension, we are free to just set it to 1
```
Comment (CarloLucibello) [inline src/layers/basic.jl:69]: why not leave `f` without type restrictions?

**Comment by DhairyaLGandhi** [comment]
Bump :)

**Comment by ToucheSir** [comment]
Since we now have `outputsize`, I think this use-case is covered.

**Comment by darsnack** [comment]
Yes, #1305 was supposed to close this.

**Pull request closed without merging**